In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
visadf= pd.read_csv('Visadataset.csv')
visadf

case_id continent education_of_employee has_job_experience  \
0         EZYV01      Asia           High School                  N   
1         EZYV02      Asia              Master's                  Y   
2         EZYV03      Asia            Bachelor's                  N   
3         EZYV04      Asia            Bachelor's                  N   
4         EZYV05    Africa              Master's                  Y   
...          ...       ...                   ...                ...   
25475  EZYV25476      Asia            Bachelor's                  Y   
25476  EZYV25477      Asia           High School                  Y   
25477  EZYV25478      Asia              Master's                  Y   
25478  EZYV25479      Asia              Master's                  Y   
25479  EZYV25480      Asia            Bachelor's                  Y   

      requires_job_training  no_of_employees  yr_of_estab  \
0                         N            14513         2007   
1                         N             2412         2002   
2                         Y            44444         2008   
3                         N               98         1897   
4                         N             1082         2005   
...                     ...              ...          ...   
25475                     Y             2601         2008   
25476                     N             3274         2006   
25477                     N             1121         1910   
25478                     Y             1918         1887   
25479                     N             3195         1960   

      region_of_employment  prevailing_wage unit_of_wage full_time_position  \
0                     West         592.2029         Hour                  Y   
1                Northeast       83425.6500         Year                  Y   
2                     West      122996.8600         Year                  Y   
3                     West       83434.0300         Year                  Y   
4                    South      149907.3900         Year                  Y   
...                    ...              ...          ...                ...   
25475                South       77092.5700         Year                  Y   
25476            Northeast      279174.7900         Year                  Y   
25477                South      146298.8500         Year                  N   
25478                 West       86154.7700         Year                  Y   
25479              Midwest       70876.9100         Year                  Y   

      case_status  
0          Denied  
1       Certified  
2          Denied  
3          Denied  
4       Certified  
...           ...  
25475   Certified  
25476   Certified  
25477   Certified  
25478   Certified  
25479   Certified  

[25480 rows x 12 columns]

In [7]:
cat_cols=visadf.select_dtypes(include='object').columns
num_cols=visadf.select_dtypes(exclude='object').columns
cat_cols
num_cols

Index(['no_of_employees', 'yr_of_estab', 'prevailing_wage'], dtype='object')

**how many ways retrieve a column**

In [ ]:
# I want to retrive 'continent' column
visadf['continent'] #series
visadf[['continent']]  #df
visadf.iloc[:,1]  #series
visadf.loc[:,['continent']]  #df
visadf.iloc[:,[1]] #df
visadf.loc[:,'continent'] #series
visadf.continent  #series

**unique**

- read the continent column apply unique method

- unique will give the continents labels which are repeating entire data

In [11]:
visadf['continent'].unique()

array(['Asia', 'Africa', 'North America', 'Europe', 'South America',
       'Oceania'], dtype=object)

In [12]:
visadf['continent'].nunique()

6

**task-1**

- we knnow only 6 labels are repeating in entire 25480 observations

- now we want to know how many time **Asia** repeating

-  we already know that category colukmn represent using frequency table

-  our main goal is to find the frequency table



In [20]:
# step 1 read continent data
#step-2 we want asia so the data equal to Asia
#5==5 o/p True
visadf['continent']
con=visadf['continent']== 'Asia'
# we are passing 'visadf['continent']== 'Asia' as condition to visadf object which get all rows with 'Asia' continent
visadf[con]
# to get lenngth that is rows of dataframe len(visadf(con))

case_id continent education_of_employee has_job_experience  \
0         EZYV01      Asia           High School                  N   
1         EZYV02      Asia              Master's                  Y   
2         EZYV03      Asia            Bachelor's                  N   
3         EZYV04      Asia            Bachelor's                  N   
5         EZYV06      Asia              Master's                  Y   
...          ...       ...                   ...                ...   
25475  EZYV25476      Asia            Bachelor's                  Y   
25476  EZYV25477      Asia           High School                  Y   
25477  EZYV25478      Asia              Master's                  Y   
25478  EZYV25479      Asia              Master's                  Y   
25479  EZYV25480      Asia            Bachelor's                  Y   

      requires_job_training  no_of_employees  yr_of_estab  \
0                         N            14513         2007   
1                         N             2412         2002   
2                         Y            44444         2008   
3                         N               98         1897   
5                         N             2339         2012   
...                     ...              ...          ...   
25475                     Y             2601         2008   
25476                     N             3274         2006   
25477                     N             1121         1910   
25478                     Y             1918         1887   
25479                     N             3195         1960   

      region_of_employment  prevailing_wage unit_of_wage full_time_position  \
0                     West         592.2029         Hour                  Y   
1                Northeast       83425.6500         Year                  Y   
2                     West      122996.8600         Year                  Y   
3                     West       83434.0300         Year                  Y   
5                    South       78252.1400         Year                  Y   
...                    ...              ...          ...                ...   
25475                South       77092.5700         Year                  Y   
25476            Northeast      279174.7900         Year                  Y   
25477                South      146298.8500         Year                  N   
25478                 West       86154.7700         Year                  Y   
25479              Midwest       70876.9100         Year                  Y   

      case_status  
0          Denied  
1       Certified  
2          Denied  
3          Denied  
5       Certified  
...           ...  
25475   Certified  
25476   Certified  
25477   Certified  
25478   Certified  
25479   Certified  

[16861 rows x 12 columns]

In [22]:
unique_labels=visadf['continent'].unique()
for i in unique_labels:
    visadf['continent']
    condition=visadf['continent']==i
    visadf[condition]
    count=len(visadf[condition])
    print(f'the number of applicants from {i} is {count}')

the number of applicants from Asia is 16861
the number of applicants from Africa is 551
the number of applicants from North America is 3292
the number of applicants from Europe is 3732
the number of applicants from South America is 852
the number of applicants from Oceania is 192


In [30]:
#create dataframe out of continents & their count
unique_labels=visadf['continent'].unique()
count=[]
for i in unique_labels:
    visadf['continent']
    condition=visadf['continent']==i
    visadf[condition]
    count.append(len(visadf[condition]))
pd.DataFrame(zip(count,unique_labels),columns=['continent','number of applicants'])


continent number of applicants
0      16861                 Asia
1        551               Africa
2       3292        North America
3       3732               Europe
4        852        South America
5        192              Oceania

In [31]:
visadf['continent'].value_counts()

continent
Asia             16861
Europe            3732
North America     3292
South America      852
Africa             551
Oceania            192
Name: count, dtype: int64